airgun/entities/contentviewfilter.py

```diff
@@ -0,0 +1,306 @@
+        """
+        view = self.navigate_to(
+            self, 'Edit',
+            cv_name=cv_name,
+            filter_name=filter_name,
+        )
+        view.content_tabs.rpms.add_rule.click()
+        view.content_tabs.rpms.table.fill([{
+            'RPM Name': rpm_name,
+            'Architecture': architecture,
+            'Version': version,
+        }])
+        view.content_tabs.rpms.table[0][4].widget.save.click()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+        # refresh as otherwise it won't be possible to read values from table
+        view.browser.refresh()
+
+    def update_package(self, cv_name, filter_name, rpm_name, new_values,
+                       architecture=None, version=None):
+        """Update package rule of RPM content view filter.
+
+        :param str cv_name: content view name
+        :param str filter_name: content view filter name
+        :param str rpm_name: existing package (RPM) name
+        :param dict new_values: dictionary with new values where keys are the
+            same as column names on UI: 'RPM Name', 'Architecture', 'Version'.
+        :param str optional architecture: filter package rule by its
+            architecture
+        :param str optional version: filter package rule by its version (string
+            value with exact correspondence to UI)
+        """
+        view = self.navigate_to(
+            self, 'Edit',
+            cv_name=cv_name,
+            filter_name=filter_name,
+        )
+        # form a dict of passed details of rpm filter
+        passed_details = {
+            'RPM Name': rpm_name,
+            'Architecture': architecture,
+            'Version': version,
+        }
+
+        def row_matches(row, **filters):
+            """Returns True if row matches all passed filters"""
+            row_values = row.read()
+            return all(
+                row_values[key] == value
+                for key, value in filters.items()
+                if value is not None
+            )
+
+        # find row which matches all filters
+        rows = [
+            row for row
+            in view.content_tabs.rpms.table.rows()
+            if row_matches(row, **passed_details)
+        ]
+        assert rows, 'Table Row not found using passed filters {}'.format(
+            passed_details)
+        row = rows[0]
+        row[4].widget.edit.click()
+        # prepare list with empty values for all preceding rows as it's
+        # impossible to fill specific table row, only in proper order
+        # todo: replace with `range(row.index)` when widgetastic.core#117 is
+        # fixed
+        values = [{} for _ in range(row.index - 1)]
+        values.append(new_values)
+        view.content_tabs.rpms.table.fill(values)
+        row[4].widget.save.click()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+        # refresh as otherwise it won't be possible to read values from table
+        view.browser.refresh()
```

Comment:
I thought about that, but decided to keep it in update since update is called rarely, when `read()` will probably occur way more ofter, and extra refresh before each `read()` will slow us down a bit, and that refresh is not really needed unless we try to add/update package rule (and only for package filter, errata/package group filters aren't affected). Still prefer moving to `read()`?

Reply:
Yeah, as it can affect new code added in future